nbclassic/static/notebook/js/notebook.js

```diff
@@ -2866,6 +2866,7 @@ define([
 
     Notebook.prototype.save_notebook_as = function() {
         var that = this;
+        var current_notebook_name = $('body').attr('data-notebook-name')
```

Comment:
Since tests are passing on `main` but not here, I suspect the code might have some issues. While I do not see it straightaway, I would add semicolons, both because lack of these can lead to unexpected evaluation in JS, and to match the style in surrounding code.

Reply:
Arrgh, I can't get the tests to pass locally on `main`, so I can't really begin to debug this branch locally either.

```bash
git clean -fdx; npm run build; OPENSSL_CONF=/etc/ssl python -m nbclassic.jstest base
```

On `main`, I still get the same issue, `Error adding link: TypeError: undefined is not a function (evaluating 'Number.isFinite(num)')`.

I'll start by cleaning this up a bit and see if we can't get a better idea of the real problem.

Reply:
I'm not exactly sure what's going on here, but it seems like there are some instances in the code base here where ES6 functions are being called, like `Math.log10` and `Math.isFinite`. When run on a version of node that doesn't support these, we get the JS errors we've been seeing in the tests. Funny thing is we haven't touched the dependencies here, so I have no idea why this branch is behaving any different in places of the code that weren't touched :thinking:

Reply:
No idea why it manifested this way, but there was a problematic arrow function buried in the click handler for the modal that prompts the user when generating the missing parent directories. Seems like tests are passing now!